### PR TITLE
Subscription-APIs: Alignment of errors with Commonalities

### LINKED ISSUE #184: Align subscription-APIs with guidelines for error scenarios
**Problem description**
Current version still does not follow the recent guidelines.

**Possible evolution**
Align codes and examples as indicated in:

https://github.com/camaraproject/Commonalities/blob/main/artifacts/camara-cloudevents/event-subscription-template.yaml

### PR DESCRIPTION
**Pull request by maxl2287**
#### What type of PR is this?

Add one of the following kinds:

* enhancement/feature

#### What this PR does / why we need it:
Alignement of errors according to Commonalities guideline.

#### Which issue(s) this PR fixes:

Fixes #184  

#### Changelog input

```
 release-note

- Aligned error scenarios list with Commonalities for subscription APIs
```

### CODE REVIEW

File: code/API_definitions/device-reachability-status-subscriptions.yaml [hunk 1]
```diff
@@ -1153,47 +1135,37 @@ components:
...
-      content:
-        application/json:
-          schema:
-            $ref: "#/components/schemas/ErrorInfo"
-          example:
-            status: 429
-            code: TOO_MANY_REQUESTS
-            message: "Endpoint does not support as many requests per time slot"
+                code: MULTIEVENT_SUBSCRIPTION_NOT_SUPPORTED
+                message: Multi event types subscription not managed
```
Comment (bigludo7) [inline code/API_definitions/device-reachability-status-subscriptions.yaml:1139]: For 422 I suggest to add also:

```
            GENERIC_422_DEVICE_IDENTIFIERS_MISMATCH:
              description: Inconsistency between device identifiers not pointing to the same device
              value:
                status: 422
                code: DEVICE_IDENTIFIERS_MISMATCH
                message: Provided device identifiers are not consistent.
            GENERIC_422_DEVICE_NOT_APPLICABLE:
              description: Service is not available for the provided device
              value:
                status: 422
                code: DEVICE_NOT_APPLICABLE
                message: The service is not available for the provided device.
```

**Comment by maxl2287** [comment]
@bigludo7 updates are done.
Please review again :)

**Review by bigludo7 — APPROVED** [review]
LGTM :)
Thanks Max.

**Comment by akoshunyadi** [comment]
I've noticed that there are quite a lot of differences to the template. Which has already differences to the design guidelines. :-) 
Which means at the end, that we don't have standardized subscriptions in Camara. 
Should we merge this PR as it is and create another one for the RC2 to match with the template, or keep it open for the RC2?

Btw. the last change of the template will be released in RC2 of the Commonalities Release, so currently we are conform with the RC1.

**Review by akoshunyadi — APPROVED** [review]
As discussed, regarding the error codes we should make one more round about the subscription template + implementations. For the RC1 it LGTM

**Pull request merged**